common/cluster/archival_status.go

```diff
@@ -44,50 +48,53 @@ const (
 	ArchivalPaused
 	// ArchivalEnabled means this cluster is currently archiving
 	ArchivalEnabled
+
+	// DomainArchivalDisabled means a domain is not configured to handle archival
+	DomainArchivalDisabled DomainArchivalStatus = iota
+	// DomainArchivalEnabled means a domain is currently archiving
+	DomainArchivalEnabled
 )
 
 // NewArchivalConfig constructs a new valid ArchivalConfig
-func NewArchivalConfig(status ArchivalStatus, defaultBucket string, enableReadFromArchival bool) *ArchivalConfig {
+func NewArchivalConfig(
+	clusterStatus ArchivalStatus,
+	enableReadFromArchival bool,
+	domainDefaultStatus DomainArchivalStatus,
+	domainDefaultURI string,
+) *ArchivalConfig {
 	ac := &ArchivalConfig{
-		status:                 status,
-		defaultBucket:          defaultBucket,
-		enableReadFromArchival: enableReadFromArchival,
+		ClusterStatus:          clusterStatus,
+		EnableReadFromArchival: enableReadFromArchival,
+		DomainDefaultStatus:    domainDefaultStatus,
+		DomainDefaultURI:       domainDefaultURI,
 	}
-	if !ac.isValid() {
-		return &ArchivalConfig{
-			status:                 ArchivalDisabled,
-			defaultBucket:          "",
-			enableReadFromArchival: false,
-		}
+	if !ac.isDomainDefaultValid() {
+		return newDisabledArchivalConfig()
 	}
 	return ac
 }
 
-// GetDefaultBucket returns the default bucket for ArchivalConfig
-func (a *ArchivalConfig) GetDefaultBucket() string {
-	return a.defaultBucket
+// ClusterConfiguredForArchival returns true if cluster is configured to handle archival, false otherwise.
+func (a *ArchivalConfig) ClusterConfiguredForArchival() bool {
```

Comment:
Keep only this method for now. More methods will be added if necessary when I update the frontend domain handler

Reply:
Makes sense.